java-checks-test-sources/src/main/java/checks/spring/TransactionalMethodVisibilityCheck.java

```diff
@@ -13,12 +15,33 @@ private interface B {
     @Transactional
     int bar(); // Compliant
   }
-  
+
+  private interface A {
+    @Async
+    int bar(); // Compliant
+  }
+
+  public static abstract class Inner {
+
+    @Async
+    protected abstract Future<String> aMethod(); // Noncompliant
+  }
+
+  @Async
+  public Future<String> asyncMethod(){ // compliant
+    return  null;
+  }
+
+  @Async
+  private  Future<String> asyncMethodPrivate(){ // Noncompliant {{Make this method "public" or remove the "@Async" annotation}}
+    return  null;
+  }
+
```

Comment:
Is there a reason why we need to place the methods within their own classes/interfaces? Can we just add them as methods of `TransactionalMethodVisibilityCheck`?